Homework 5

korejur opened the pull request:
nisam baš uspjela ovaj dio s testom mutate funkcije kod gumba za brisanje. nije mi implementiran kao što je bilo u predavanju možda je do toga.

uglavnom htjela bih da to prokomentiramo na konzultacijama možda se može brzo riješiti samo ja ne znam sad kako. probala sam nešto s tim unutar ShowContainerReviewDeletion.test.tsx

korejur committed:
task 1

register and login

korejur committed:
task 2

korejur committed:
task 3

migrate api

korejur committed:
Update page.tsx

korejur committed:
Merge branch 'main' into homework-4

korejur committed:
Revert "Merge branch 'main' into homework-4"

This reverts commit 79eb7af8ea53575dc09ab383b4b72f1be7b75971, reversing
changes made to 0aef943b5f6a8e786b3368f9108793c243428c4e.

korejur committed:
made changes

sidebar, types, logout redirect

korejur committed:
task 1

create and delete review

korejur committed:
task 2

tests, 1 failed

korejur committed:
test update

ovaj test bi trebao provjeravati da se dogodi mutate nakon što se obriše review, ali ne radi i ne znam kako popraviti :/

meni taj gumb za brisanje nije bas implementiran kao na predavanjima

darrac commented on tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx:
```diff
@@ -1,101 +1,67 @@
+    try {
+      await createReview(swrKeys.create_review, { arg: review });
+      console.log(reviewListResponse);
+      mutate(swrKeys.reviews(Number(showId)));
```
This `mutate` only changes the values in you internal SWR cache.
What you need to trigger a POST request to the server is `useSwrMutation` and its `trigger` function.

Also, you can then move this directly to the component that triggers, in this case, `ReviewForm`

darrac commented on tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx:
```diff
@@ -1,101 +1,67 @@
     try {
-      return JSON.parse(reviewListString);
+      await deleteReview(swrKeys.review(Number(reviewToRemove.id)));
+      mutate(swrKeys.reviews(Number(showId)));
```
Same comment as for the add review. You can find an example of this in the lecture code.

darrac reviewed (changes requested):
You need to update the way you use useSwr and useSwrMutation as follows:
1.  When you need to fetch data, use `useSwr`
2. When you need to POST, PATCH or DELETE something on the server, use `useSwrMutation`
3. After you change/add/delete some entities on the server with `useSWRMutation` tell all the components that render a list of those entities to update by calling `mutate`.

korejur committed:
ReviewItem test, relocated triggers

promijenila sam lokaciju triggera za (brisanje i) dodavanje, obrisala one testove i napravila novi koji u sebi sadrzi sve potrebno.
tak da valjda sad sve radi

darrac commented on tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx:
```diff
@@ -1,31 +1,22 @@
   return (
     <>
-      <ReviewForm onAdd={handleAddReview} />
+      <ReviewForm showId = {showId} />
```
Why not fetch the reviews in this component?

korejur replied:
Ako raspodijelim fetchanje na ShowReviewSection i ShowDetailsCard onda više u ShowContaineru ne mogu provjeriti isLoading i error? Nakon te raspodjele, loading i error se provjeravaju u svakoj komponenti zasebno, ali onda mi se dvaput prikaže poruka Loading...
Što mi je činiti?

darrac replied:
To ti sada ne izgleda lijepo zato sto ti UI pokaze obicni `Loading` tekst. 
U stvarnom slucaju bi koristila https://v2.chakra-ui.com/docs/components/skeleton ili nesto slicno pa bi to OK izgledalo.
Nemoj se sada brinuti s tim sto se loading prikazuje 2 puta.

korejur replied:
Okej, pushala sam to. Nisam stigla sad u apartmanu primijenit to i u Homework 6 jer smo u nekoj žurbi bili, ali ako hoćeš, možeš ti to mergati. Jučer sam radila i na popravcima iz Homework 6 tako da bi to trebalo bit to kad se spoji s ovim najnovijim dijelom.

korejur replied:
Riješeno :)

darrac commented on tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx:
I don't think there is a need to calculate average rating anymore as you receive it from the API as part of the show resource.

korejur committed:
removed avgRating function

korejur committed:
change average rating

on review add/delete call mutate

korejur committed:
Update ReviewItem.tsx

korejur committed:
moved fetching
